src/modules/0.0.14/aws_elb/sql/create_custom_constraints.sql

```diff
@@ -0,0 +1,14 @@
+create or replace function check_load_balancer_availability_zones(_load_balancer_name varchar, _availability_zones varchar[]) returns boolean
+language plpgsql security definer
+as $$
+declare
+  _number_of_records integer;
+begin
+  select count(*) from availability_zone where name = any(_availability_zones) into _number_of_records;
+  if _number_of_records != array_length(_availability_zones, 1) then
+    raise exception 'Load balancer % includes one or more invalid availability zones', _load_balancer_name;
+  end if;
+  return true;
+end;
+$$;
+ALTER TABLE load_balancer ADD CONSTRAINT check_load_balancer_availability_zones CHECK (check_load_balancer_availability_zones(load_balancer_name, availability_zones));
```

Comment:
since this check will be called on insert, would not be possible that theres no `load_balancer_name`, would that constraint fail first that this one?

Reply:
It shouldn't because the `any(NULL)` should return no records to count and the length of it is also 0.

Reply:
Oh I was meaning since the load balancer name is being passed as an argument, but it is true that if nor there is `NULL` so all good